vignettes/wwinference.Rmd

```diff
@@ -297,33 +309,62 @@ fit <- wwinference::wwinference(
     infection_feedback_pmf = infection_feedback_pmf,
     params = params
   ),
-  mcmc_options = get_mcmc_options(),
+  fit_opts = get_mcmc_options(iter_warmup = 100, iter_sampling = 100),
   compiled_model = model
 )
 ```
 
-# The `wwinference` object
-The `wwinference()` function returns a `wwinference` object which includes
-a `draws_df`, the underlying `CmdStan` object (`raw_fit_obj`), and three
-"spines" that map the stan indices to the data which include:a
-`date_time_spine`, `lab_site_spine`, and `subpop_spine`. The `draws_df` is
-intended to provide an easy to work with tibble of posterior draws of
-the estimated, nowcasted, and forecasted expected observed hospital admissions
-and wastewater concentrations, as well as the latent variables of interest
-including the site-level R(t) estimates and the state-level R(t) estimate.
-```{r}
-head(fit)
+# The `wwinference_fit` object
+
+The `wwinference()` function returns a `wwinference_fit` object which includes
+the underlying and the underlying
+[`CmdStanModel` object](https://mc-stan.org/cmdstanr/reference/CmdStanModel.html)
+ (`fit`), a list of the two sources of input
+data (`input_data`), the list of the arguments passed to stan
+(`stan_data_args`), and the list of the MCMC options (`mcmc_options`) passed to
+stan. We show how to generate downstream elements from a `wwinference_fit`
+object.
+
+## Extracting the posterior predictions
+
+Working with the posterior predictions alongside the input data can be useful
+to check that your model is fitting the data well and that the
+nowcasted/forecast quantities look reasonable.
+
+We will generate a dataframe that we'll call `draws_df`, that contains
+the posterior draws of the estimated, nowcasted, and forecasted expected
+observed hospital admissions and wastewater concentrations, as well as the
+latent variables of interest including the site-level R(t) estimates and the
+state-level R(t) estimate.
+
+We can generate this directly on the output of `wwinference()` using:
+```{r extracting-draws}
+draws_df <- get_draws_df(ww_fit)
 ```
-# Summarizing and plotting the model fit
-The `draws_df` object is intended to be used to easily plot relevant model
-outputs against data. This can be useful to get a sense of if your model is
-fitting the data well, and if the nowcasted/forecast quantities look reasonable.
 
-We can create the plots using the fitting wrapper functions (figures can
-also created directly using the `draws_df`)
+### Using explicit passed arguments rather than S3 methods
+
+Rather than using S3 methods supplied for `wwinference()`, the elements in the
+`wwinference_fit` object can also be used directly to create this dataframe.
+This is demonstrated below:
+
+```{r extracting-draws-explicit}
+draws_df_explicit <- get_draws_df(
```

Comment:
@gvegayon Question for my understanding -- I thought that this would call the `.default` method, since we haven't specified the `.dataframe` method. Wouldn't that mean that this should throw an error?

Reply:
We now have a data.frame method ;).